### PR TITLE
fix: #414 -  changed get request to post

### PR DESCRIPTION
**Pull request by AshAman999**
### What
- migrating all the get requests to post

### Screenshot
(Insert a screenshot to provide visual record of your changes, if visible)
### Fixes bug(s)
- #354 and #414 
### Part of 
- #407 
(please be as granular as possible)

### CODE REVIEW

**Comment by AshAman999** [comment]
@monsieurtanuki  I have done the task,please review the changes,
It's my 1st  pr in this organization so sorry if any incompleteness is found in the code.

File: lib/openfoodfacts.dart [hunk 1]
```diff
@@ -300,13 +301,13 @@ class OpenFoodAPIClient {
   }) async {
     var searchUri = UriHelper.getUri(
       path: '/cgi/search.pl',
-      queryParameters: configuration.getParametersMap(),
       queryType: queryType,
```
Comment (monsieurtanuki) [inline lib/openfoodfacts.dart:304]: What about `addUserAgentParameters: false,`?

**Review by monsieurtanuki — COMMENTED** [review]
It looks OK @AshAman999 except that for some reason the unit tests did not start.
Besides, you missed at least one `addUserAgentParameters: false`: I suggest that you create a new method `static Uri UriHelper.getPostUri(final String path, {final QueryType? queryType,})` and use it in at least the 15 times where `UriHelper.getUri` is used.

**Comment by AshAman999** [comment]
@monsieurtanuki  please have a look, i guess this would be okay now

**Review by monsieurtanuki — CHANGES_REQUESTED** [review]
Hi @AshAman999!
The code looks rather OK, but have a look at my comments: there are things to fix.
Anyway the tests didn't pass: don't forget to run the tests locally (`dart test`) before updating your PR.

**Review by monsieurtanuki — APPROVED** [review]
Looks good now, thank you @AshAman999!

**Comment by monsieurtanuki** [comment]
Congratulations @AshAman999 for your first PR here!

**Pull request merged**